Core/MIPS/MIPSInt.cpp

```diff
@@ -974,7 +974,14 @@ namespace MIPSInt
 		{
 		case 0: F(fd) = F(fs) + F(ft); break; // add.s
 		case 1: F(fd) = F(fs) - F(ft); break; // sub.s
-		case 2: F(fd) = F(fs) * F(ft); break; // mul.s
+		case 2: // mul.s
+			if ((my_isinf(F(fs)) && F(ft) == 0.0f) || (F(fs) == 0.0f && my_isinf(F(ft)))) {
```

Comment:
It's probably better to always have the my_isinf check on the left side of the & clauses - inf is a lot rarer than 0.0 so more short-circuiting.

(by the way, have you checked the NaN sign of negative inf * positive zero and positive inf * negative zero?

Reply:
Fair, changed.

Yes, the sign and order of operands make no difference, neither does the flush to zero flag.

-[Unknown]